src/main/docker/bootstrap.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+cd /opt/Artemis
+
+if [ -z "$(ls -A config)" ]; then
+   echo "Config is Empty .. copying default ones .."
+   cp -n -a /defaults/Artemis/. config/
+else
+   echo "Config is not empty .. not copying default configs .."
+fi
+
+# Create sync file that indicates waiting for services
+mkdir -p sync
+touch sync/waiting-backend
+
+# Ensure at least the directories are owned by artemis. "-R" takes too long
+chown artemis:artemis config data sync
+
+# Allow waiting for other services
+if [ -n "${WAIT_FOR}" ]; then
+  hosts_ports=$(echo $WAIT_FOR | tr "," "\n" | tr -d "\"")
+  for host_port in $hosts_ports
+  do
+    until [[ "$(curl -s -o /dev/null -L -w ''%{http_code}'' http://$host_port)" == "200" ]]
+    do
+      echo "Waiting for $host_port"
+      sleep 5
+    done
+  done
+fi
```

Comment:
Again, I don't like the introduced complexity here. If the application crashes, the container should crash as well. To ensure that containers start in the correct order and in time I suggest the `depends_on` ([ref](https://docs.docker.com/compose/compose-file/compose-file-v3/#depends_on)) and `restart_policy`  ([ref](https://docs.docker.com/compose/compose-file/compose-file-v3/#restart_policy)) directives.

Reply:
> If the application crashes, the container should crash as well.

Agree, but it will crash :)

> To ensure that containers start in the correct order and in time I suggest the depends_on ([ref](https://docs.docker.com/compose/compose-file/compose-file-v3/#depends_on))

depends_on is not sufficient. Since the container depends on gitlab & jenkins (and they need time to spin up).
I have the strong opinion that this is needed :)

Reply:
And can you provide me with an example where Artemis crashes because gitlab is not running? I can't reproduce that right now

Reply:
In past versions, artemis crashed during the  update of the internal artemis user (because the update could not propagated).
I'll check whether this is the case today.


> I'm not sure what you mean. The [restart policy](https://docs.docker.com/config/containers/start-containers-automatically/#use-a-restart-policy) will restart the container if the application inside crashes.

I just wanted to say, that the bootstrap.sh file does not contain a restart mechanism anymore (supervisord was removed :) )

Reply:
Sorry @Mtze .. pasted to the wrong place :) 

See https://github.com/ls1intum/Artemis/pull/4886#discussion_r840447025

Reply:
Ok @Mtze @sleiss let's discuss in the correct thread :)

If the exception can be ignored, I think we can drop the section of the script

Reply:
If you remove this section, I guess the PR can be merged

Reply:
Done